### ISSUE TITLE
Test on MacOS Big Sur and Monterey

### ISSUE THREAD

**Issue by RhetTbull** [RhetTbull/macnotesapp#1]

**Comment by chadmando** [comment]
I have Big Sur 11.7.9 installed on a daily driver with a lot of Notes.  If you want any help with testing on that OS version.  Let me know if I can help.

**Comment by RhetTbull** [comment]
@chadmando thanks! I'm working on another project at the moment but want to get back to macnotesapp soon as I've got some things I want to implement. Will definitely appreciate the help testing.

**Comment by chadmando** [comment]
I walked through the tests this morning using `pytest -v -s tests/` using python 3.11 on Big Sur.  The only failures that I see all relate to Note count/length.  The test's counts were all +2 greater than the Notes app's counts. This included both collectively (iCloud + On My Mac) and the iCloud only count.  I didn't see the On My Mac count to determine if it was something only related to the iCloud count.

Disclaimer: I have been moving a LOT of notes from Evernote to Notes app and I have found some "dead notes" in that process as I add them to iCloud.  I will walk through my most recent imports and double check all are valid. 

failed tests for reference:
```text
=============================================================== short test summary info ================================================================
FAILED tests/test_applescript_interface.py::test_notes_len - AssertionError: assert False
FAILED tests/test_applescript_interface.py::test_notes_notes - AssertionError: assert False
FAILED tests/test_applescript_interface.py::test_notes_generator - AssertionError: assert False
FAILED tests/test_applescript_interface.py::test_account - AssertionError: assert False
================================================= 4 failed, 19 passed, 1 skipped in 945.97s (0:15:45) ==================================================
```

**Comment by RhetTbull** [comment]
Thank you for taking time to do this!  On the note counts, did you include "Recently Deleted"? Those are part of the overall count. On Ventura my counts match when including Recently deleted (and any other accounts, like Google, that have notes enabled). 

If your counts are still off with recently deleted I'll take a look to see if I can debug this. I don't have access to Big Sur but do have a Monterey virtual machine I can use.

**Comment by RhetTbull** [comment]
@all-contributors please add @chadmando for user testing

**Comment by chadmando** [comment]
> Thank you for taking time to do this! On the note counts, did you include "Recently Deleted"? Those are part of the overall count. On Ventura my counts match when including Recently deleted (and any other accounts, like Google, that have notes enabled).
> 
> If your counts are still off with recently deleted I'll take a look to see if I can debug this. I don't have access to Big Sur but do have a Monterey virtual machine I can use.

Good call.  It was the Recently Deleted causing my confusion.  Counts match when I include those.

**Comment by chadmando** [comment]
@RhetTbull have you tested on Sonoma? I have access to Sonoma and can test sometime in the near future.

**Comment by RhetTbull** [comment]
@chadmando I've tested on the Sonoma beta which I installed in a VM and everything works fine. I'd welcome test feedback from the released version of Sonoma. On my two personal Macs I tend to lag 1 major version.  I've now got an M1 Mini and virtualization works well on the Apple Silicon so I've been able to create VMs for additional testing.
